### DISCUSSION TITLE
Netbeans 12.6 JavaFX2 will not activate in BSD

### DISCUSSION THREAD

**Discussion by Astralnut** [apache/netbeans#3815] [General]
Here is the error I get;
Activation failed: Cannot enable JavaFX 2 Support[org.netbeans.modules.javafx2.kit] - module org.netbeans.api.htmlui > 1.15

I'm just trying to set Netbeans for Java but it won't install the FX plugin. That makes it impossible to follow the porting guide for FX in 12.6.
Any help is appreciated.
Astralnut

**Comment by mbien** [comment]
have you tried NetBeans 13?

**Reply by Astralnut** [reply]
Thank you for the suggestion. I did not know there was a version 13. I got it installed the FX module activated and all looked good. Then when I created a project, it said  "The JDK your using in not FX capable". I am using open jdk17. 
I admit I'm new to setting this up. Is there another version of the JDK I should use?
Thanks again;
Astralnut

**Reply by mbien** [reply]
i have to admit, i never used the JavaFX plugin so far. You can use JavaFX in normal java maven projects, all you have to do is to register the dependency and a launcher plugin in the maven project pom. You should find many guides if you search javafx + maven for a initial setup.

**Reply by mbien** [reply]
I just tried the following:
new project -> java with maven -> Simple JavaFX Maven Archetype (Gluon) -> next...

It created a minimal JavaFX maven project. It ran fine on JDK 17.

I recommend incrementing the javafx version manually in the project pom after the project is created, since the default is fairly old. We should probably fix that. (but even the old version worked on JDK 17)

**Reply by Astralnut** [reply]
Hello;
I tried the maven like you suggested and that worked. Thank you again. I got some reading to do on maven but that does look like a good project management tool for development. 
Cheers and thanks again!
Astralnut
